app/src/main/java/com/jerboa/util/MarkwonSpoilerPlugin.kt

```diff
@@ -0,0 +1,140 @@
+            val spoilerTitleRegex = Regex("(:::\\s*spoiler\\s*)(.*)")
+            // Find all spoiler "start" lines
+            val spoilerTitles = spoilerTitleRegex.findAll(text)
+
+            for (match in spoilerTitles) {
+                val spoilerTitle = match.groups[2]!!.value
+                visitor.builder().setSpan(
+                    SpoilerTitleSpan(spoilerTitle),
+                    start,
+                    start + match.groups[2]!!.range.last,
+                    Spanned.SPAN_EXCLUSIVE_EXCLUSIVE,
+                )
+            }
+
+            val spoilerCloseRegex = Regex("^(?!.*spoiler).*:::")
+            // Find all spoiler "end" lines
+            val spoilerCloses = spoilerCloseRegex.findAll(text)
+            for (match in spoilerCloses) {
+                visitor.builder()
+                    .setSpan(SpoilerCloseSpan(), start, start + 3, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)
+            }
+        }
+    }
+
+    override fun afterSetText(textView: TextView) {
+        try {
+            val spanned = SpannableStringBuilder(textView.text)
+            val spoilerTitleSpans =
+                spanned.getSpans(0, spanned.length, SpoilerTitleSpan::class.java)
+            val spoilerCloseSpans =
+                spanned.getSpans(0, spanned.length, SpoilerCloseSpan::class.java)
+
+            spoilerTitleSpans.sortBy { spanned.getSpanStart(it) }
+            spoilerCloseSpans.sortBy { spanned.getSpanStart(it) }
+
+            spoilerTitleSpans.forEachIndexed { index, spoilerTitleSpan ->
+                val spoilerStart = spanned.getSpanStart(spoilerTitleSpan)
+
+                var spoilerEnd = spanned.length
+                if (index < spoilerCloseSpans.size) {
+                    val spoilerCloseSpan = spoilerCloseSpans[index]
+                    spoilerEnd = spanned.getSpanEnd(spoilerCloseSpan)
+                }
+
+                var open = false
+                val getSpoilerTitle = { openParam: Boolean ->
+                    if (openParam) "▼ ${spoilerTitleSpan.title}" else "▶ ${spoilerTitleSpan.title}"
+                }
+
+                val spoilerTitle = getSpoilerTitle(false)
+
+                val spoilerContent = spanned.subSequence(
+                    spanned.getSpanEnd(spoilerTitleSpan) + 1,
+                    spoilerEnd - 3,
+                ) as SpannableStringBuilder
+
+                // Remove spoiler content from span
+                spanned.replace(spoilerStart, spoilerEnd, spoilerTitle)
+                // Set span block title
+                spanned.setSpan(
+                    spoilerTitle,
+                    spoilerStart,
+                    spoilerStart + spoilerTitle.length,
+                    Spanned.SPAN_EXCLUSIVE_EXCLUSIVE,
+                )
+
+                val wrapper = object : ClickableSpan() {
+                    override fun onClick(p0: View) {
+                        if (!enableInteraction) {
```

Comment:
Does the value of this need to be flipped too?

Reply:
Yeah, sorry about that